core/node/events/miniblock.go

```diff
@@ -141,11 +140,13 @@ func (b *MiniblockInfo) asStorageMbWithData(bytes []byte) *storage.WriteMinibloc
 func (b *MiniblockInfo) forEachEvent(op func(e *ParsedEvent, minibockNum int64, eventNum int64) (bool, error)) error {
 	blockNum := b.header().MiniblockNum
 	eventNum := b.header().EventNumOffset
-	for _, event := range b.events() {
-		c, err := op(event, blockNum, eventNum)
-		eventNum++
-		if !c {
-			return err
+	if b.Proto != nil {
```

Comment:
this is not correct:

if MiniblockInfo was parsed with DontParseEvents, forEachEvent should fail (`events()` panics in this case, but it's possible to return Err_INTERNAL if this is more desirable).

in other words, it shouldn't silently pretend that events were processed correctly.